collect_app/src/main/java/org/odk/collect/android/receivers/NetworkReceiver.java

```diff
@@ -80,9 +80,7 @@ public void onReceive(Context context, Intent intent) {
             if (ni == null || !ni.isConnected()) {
                 // not connected, do nothing
             } else {
-                if (interfaceIsEnabled(context, ni)) {
-                    uploadForms(context);
-                }
+                uploadForms(context, ni);
```

Comment:
Instead of passing the network interface in here, I think it would be better to pass in the result of `interfaceIsEnabled`. That value doesn't change across all the forms so it's better if it doesn't have to be recomputed.